src/gui/sdlmain.cpp

```diff
@@ -4616,13 +4620,16 @@ static int erase_mapper_file()
 	return 0;
 }
 
+#if !(SDL_VERSION_ATLEAST(3, 0, 0))
 static void override_wm_class()
 {
 #if !defined(WIN32)
 	constexpr int overwrite = 0; // don't overwrite
-	setenv("SDL_VIDEO_X11_WMCLASS", DOSBOX_PROJECT_NAME, overwrite);
+	setenv("SDL_VIDEO_X11_WMCLASS", DOSBOX_APP_ID, overwrite);
+	setenv("SDL_VIDEO_WAYLAND_WMCLASS", DOSBOX_APP_ID, overwrite);
 #endif
 }
+#endif
```

Comment:
In this function, you need to say if we're using SDL 3, call `SDL_SetHint(SDL_HINT_APP_ID, DOSBOX_APP_ID)` ,else do the existing block.  The SDL 3 hint can probably be run for all platforms.  SDL documentation only mentioned *nix for this hint but they might do something with this on other platforms by the time SDL 3 ships and it shouldn't hurt anything.

We should also probably exclude MacOS from the X11/Wayland block while we're at it.